utils/esr_peer_info.cpp

```diff
@@ -0,0 +1,382 @@
+                    if (peer_id.size() >= peer_id_length)
+                    {
+                        trim_right_nulls(peer_id);
+                        results.emplace_back(peer_id);
+                        peer_id = "";
+                        state = PEER_ID_SIZE;
+                    }
+                default:
+                    break;
+            }
+            index++;
+        }
+        assert(results.size()==node_count);
+        return results;
+    }
+
+
+    // TODO: replace this with a function that uses the ABI to parse the response
+    bzn::peer_address_t
+    parse_get_peer_info_result_to_peer_address(const std::string &peer_id, const std::string_view &result)
+    {
+        uint16_t    port{0};
+        std::string host;
+        uint16_t    http_port{0};
+        std::string name;
+        enum {NODE_COUNT, NA_0, HTTP_PORT, NA_1, NODE_PORT, NA_2, NODE_HOST, NA_3, NODE_NAME} state {NODE_COUNT};
+
+        // NOTE: I needed to add the extra null character to "line" so that any 64 length buffer read from the result
+        // will be correctly null terminated for the boost::algorithm::unhex function
+        char line[ESR_RESPONSE_LINE_LENGTH + 1]{0};
+        std::istringstream stream{result.begin()};
+        while(stream.read(line, ESR_RESPONSE_LINE_LENGTH))
+        {
+            switch (state)
+            {
+                case NODE_COUNT: state = NA_0; break;
+                case NA_0: state = HTTP_PORT; break;
+                case HTTP_PORT:
+                    http_port = uint16_t(std::strtoul(line, nullptr, 16));
+                    if (0 == http_port)
+                    {
+                        LOG(warning) << "Invalid value for http port:[" << http_port << "] node may not exist";
+                    }
+                    state = NA_1;
+                    break;
+                case NA_1: state = NODE_PORT; break;
+                case NODE_PORT:
+                    port = std::strtoul(line, nullptr, 16);
+                    if (0 == http_port)
+                    {
+                        LOG(warning) << "Invalid value for port:[" << port << "] node may not exist";
+                    }
+                    state = NA_2;
+                    break;
+                case NA_2:
+                    state = NODE_HOST;
+                    break;
+                case NODE_HOST:
+                    host = hex_to_char_string(line);
+                    trim_right_nulls(host);
+                    if (0 == http_port)
+                    {
+                        LOG(warning) << "Empty value for host ip, node may not exist";
+                    }
+                    state = NA_3;
+                    break;
+                case NA_3:
+                    state = NODE_NAME;
+                    break;
+                case NODE_NAME:
+                    name = hex_to_char_string(line);
+                    trim_right_nulls(name);
+                    if (0 == http_port)
```

Comment:
same for all others... it's sufficient to do:

if (http_port)
...

I know why you do this, but it's really not needed.

Reply:
Cool, done.